firmware/Core/Src/telecommands/mpi_telecommand_defs.c

```diff
@@ -0,0 +1,76 @@
+#include "telecommands/telecommand_args_helpers.h"
+#include "telecommands/mpi_telecommand_defs.h"
+#include "mpi/mpi_command_handling.h"
+#include "transforms/arrays.h"
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <inttypes.h>
+
+
+///@brief Send a configuration command & params (IF ANY) to the MPI encoded in hex
+/// @param args_str 
+/// - Arg 0: Hex-encoded string representing the configuration command + params (IF ANY) being sent to the MPI
+/// @param tcmd_channel The channel on which the telecommand was received, and on which the response should be sent
+/// @param response_output_buf The buffer to write the response to
+/// @param response_output_buf_len The maximum length of the response_output_buf (its size)
+/// @return 0: if successful, 1: if invalid input, 2: error transmitting, 3: MPI failed to execute cmd
+uint8_t TCMDEXEC_mpi_send_command_hex(const char *args_str, TCMD_TelecommandChannel_enum_t tcmd_channel,
+                                      char *response_output_buf, uint16_t response_output_buf_len)
+{
+    // Get the length of the input string
+    size_t args_str_len = strlen(args_str);
+
+    // Validate command argument
+    if (args_str == NULL)
+    {
+        snprintf(response_output_buf, response_output_buf_len, "No commands received. Sample MPI command: 0204, sets HV_INNER_DOME_SCAN_MODE(02) to sawtooth with beam tracking (04)");
+        return 1; // Error code: NULL input
+    }
+
+    if (args_str_len % 2 != 0)
+    {
+        snprintf(response_output_buf, response_output_buf_len, "Invalid hex-encoded string length. Must be even.\n");
+        return 2; // Error code: Invalid hex input
+    }
+
+    // Parse hex-encoded string to bytes
+    uint16_t args_bytes_size = args_str_len/2;
+    uint16_t args_bytes_len;
+    uint8_t args_bytes[args_bytes_size];
+    uint8_t bytes_parse_result = TCMD_extract_hex_array_arg(args_str, 0, args_bytes, args_bytes_size, &args_bytes_len);
+
+    // Check for hex parsing errors
+    if(bytes_parse_result != 0){
+        snprintf(response_output_buf, response_output_buf_len, "Invalid character found in hex-encoded string");
```

Comment:
This is actually "Invalid hex argument received". There are other cases that cause it to be invalid (other than just "invalid character").